vir/src/legacy/ast/expr.rs

```diff
@@ -1140,11 +1140,11 @@ impl Expr {
             Expr::Local(LocalVar { ref typ, .. }, _)
             | Expr::Variant(_, Field { ref typ, .. }, _)
             | Expr::Field(_, Field { ref typ, .. }, _)
-            | Expr::AddrOf(_, ref typ, _)
-            | Expr::LetExpr(LocalVar { ref typ, .. }, _, _, _) => typ,
+            | Expr::AddrOf(_, ref typ, _) => typ,
             Expr::LabelledOld(_, box ref base, _)
             | Expr::Unfolding(_, _, box ref base, _, _, _)
-            | Expr::UnaryOp(_, box ref base, _) => base.get_type(),
+            | Expr::UnaryOp(_, box ref base, _)
+            | Expr::LetExpr(_, _, box ref base, _) => base.get_type(),
```

Comment:
This one was annoying to find. @vakaras are `let`s assigned the correct type (i.e. the type of the `base` expression, *not* the type of the bound variable) in the new layers?

Reply:
Shouldn't the types match exactly? We probably should add an assert that ensures that.

Reply:
`let foo == 1 in (1 + 1 == 2)`: the let-expression has type `Bool`, but the bound variable is `Int`.

Reply:
Oh, I see.